### DISCUSSION TITLE
raspapd.service can not start dhcpcd.service due to it is not found

### DISCUSSION THREAD

**Discussion by velison1-mbed** [RaspAP/raspap-webgui#1422] [General]
Hello,

I am trying to use as an access point my raspberry pi zero w. I started with fresh 32 bit lite raspberry pi image. I imaged by rpi imager and install raspap with quick installer and reboot the pi. When I check the wi-fi in my phone, I see raspap-webgui network and tried to connect it but I can't. My phone can not get an ip address. When I check the services (dnsmasq, hostapd and raspapd) I realized that raspapd.service can not start dhcpcd.service. These are raspapd.service logs

-- Boot 1cfd9105d7fd4a648d7ab3e408a433e8 --
Oct 17 08:54:40 raspberrypi systemd[1]: Starting raspapd.service - RaspAP Service Daemon...
Oct 17 08:54:41 raspberrypi bash[805]: Stopping network services...
Oct 17 08:55:10 raspberrypi bash[805]: Changing RaspAP Daemon --interface to uap0
Oct 17 08:55:11 raspberrypi bash[805]: Disabling systemd-networkd
Oct 17 08:55:21 raspberrypi bash[805]: Starting network services...
Oct 17 08:55:25 raspberrypi systemctl[881]: Failed to start dhcpcd.service: Unit dhcpcd.service not found.
Oct 17 08:55:39 raspberrypi bash[805]: RaspAP service start DONE
Oct 17 08:55:39 raspberrypi systemd[1]: raspapd.service: Deactivated successfully.
Oct 17 08:55:39 raspberrypi systemd[1]: Finished raspapd.service - RaspAP Service Daemon.

After that I tried to install dhcpcd with the command "sudo apt-get install dhcpcd5" even that didn’t help. Because there is no dhcpcd.service. There is dhcpcd.conf but no dhcpcd.service.

Also these are hostapd.service logs while the phone trying to connect
Oct 17 09:46:38 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: disassociated
Oct 17 09:46:38 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: disassociated
Oct 17 09:46:51 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: associated
Oct 17 09:46:51 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 RADIUS: starting accounting session 79CCD6527C478A6F
Oct 17 09:46:51 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 WPA: pairwise key handshake completed (RSN)
Oct 17 09:47:27 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: disassociated
Oct 17 09:47:27 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: disassociated
Oct 17 09:47:31 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 IEEE 802.11: associated
Oct 17 09:47:31 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 RADIUS: starting accounting session 4D16C26D31DDA6F4
Oct 17 09:47:31 raspberrypi hostapd[1093]: wlan0: STA 04:b1:67:af:0f:a5 WPA: pairwise key handshake completed (RSN)

How can I fix it? Can you help me?

**Comment by Ekiben542** [comment]
What is the output of systemctl status dhcpd?

**Comment by velison1-mbed** [comment]
> What is the output of systemctl status dhcpd?

Unit dhcpcd.service not found.

**Comment by Ekiben542** [comment]
Hmmm, that's a tough one...
If you have an /etc/systemd/system/dhcpcd.service.d
directory, you will find wait.conf there.
[Service].
ExecStart
ExecStart=/usr/sbin/dhcpcd -w -q
I think it would be possible to do this by writing .
If there is no dhcpcd.service.d directory at all, I think you can create a dhcpcd.service.d directory, create a wait.conf and put the above in it. (At your own risk.))

**Comment by velison1-mbed** [comment]
> Hmmm, that's a tough one... If you have an /etc/systemd/system/dhcpcd.service.d directory, you will find wait.conf there. [Service]. ExecStart ExecStart=/usr/sbin/dhcpcd -w -q I think it would be possible to do this by writing . If there is no dhcpcd.service.d directory at all, I think you can create a dhcpcd.service.d directory, create a wait.conf and put the above in it. (At your own risk.))

![image](https://github.com/RaspAP/raspap-webgui/assets/147369417/111790da-ce1c-46e8-9282-7461aa77d5e9)

should I create a new directory in here named dhcpcd.service.d?

**Comment by Ekiben542** [comment]
Yes

**Comment by Ekiben542** [comment]
Then, if you create wait.conf there and write what you wrote earlier, it will work (Maybe)

**Comment by billz** [comment]
dhcpcd is the default DHCP client used by RPi OS. If it isn't present on your system, or the systemd service doesn't exist, this is not a clean install.

```
$ which dhcpcd
/usr/sbin/dhcpcd

$ systemctl status dhcpcd.service
● dhcpcd.service - DHCP Client Daemon
     Loaded: loaded (/lib/systemd/system/dhcpcd.service; enabled; vendor preset: enabled)
```

You can save yourself some trouble by starting with a fresh OS.

**Reply by velison1-mbed** [reply]
When I check the release notes of the last raspberry os image I realized that network manager is used instead of dhcpcd anymore. I think the main problem is this. Because I don't even remember how many times I rewrote the image. It can't be a coincidence that dhcpcd isn't present in all of them.

You can check it [here](https://downloads.raspberrypi.com/raspios_lite_armhf/release_notes.txt?_gl=1*1qbpetn*_ga*MTU5MTUyNjAzLjE2OTcxOTk5NDU.*_ga_22FD70LWDS*MTY5NzYzMTA2Mi4zLjEuMTY5NzYzMTA2OC4wLjAuMA..), it is reported in the seventh article of the latest version.

![image](https://github.com/RaspAP/raspap-webgui/assets/147369417/a5772059-8fac-4f17-a27a-2b2c560d4c51)


NetworkManager used instead of dhcpcd as networking interface; various changes made to networking plugin to support this

**Reply by billz** [reply]
Debian 12 (bookworm) was released just last week. This project currently supports Bullseye
https://docs.raspap.com/#compatible-operating-systems

**Reply by velison1-mbed** [reply]
Yes, you are right. I realized it a bit late :sweat_smile: 

Thanks for your support.

**Reply by billz** [reply]
I'll test an installation with bookworm shortly 🙂

**Reply by billz** [reply]
The installer already has support for Debian 12. I ran through a clean install with 64-bit RPi OS (bookworm) and observed the following.

during install:
```
Setting up dhcpcd (1:9.4.1-24~deb12u2) ...
Created symlink /etc/systemd/system/multi-user.target.wants/dhcpcd.service → /lib/systemd/system/dhcpcd.service.
```
post-install:
```
sudo systemctl status dhcpcd.service 
● dhcpcd.service - DHCP Client Daemon on all interfaces
     Loaded: loaded (/lib/systemd/system/dhcpcd.service; enabled; preset: enabled)
     Active: active (running) since Thu 2023-10-19 19:17:14 CEST; 1min 6s ago
```

```
dhcpcd --version
dhcpcd 9.4.1
Copyright (c) 2006-2021 Roy Marples
Compiled in features: INET ARP ARPing IPv4LL INET6 DHCPv6 AUTH PRIVSEP
```

```
uname -a
Linux raspberrypi 6.1.0-rpi4-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.54-1+rpt2 (2023-10-05) aarch64 GNU/Linux
```

Suggest you try 64-bit if possible for now. I'll see if I can repro this on the 32-bit bookworm distro.

**Reply by billz** [reply]
@velison1-mbed you're correct, the 32-bit bookworm distro reports `Raspbian GNU/Linux 12` rather than Debian. This causes the installer to skip installation of dhcpcd. I'll push a fix shortly. Thanks for raising this!

**Reply by billz** [reply]
Should be resolved in #1423

**Comment by billz** [comment]
The installer adds support for 32-bit Raspbian 12 (bookworm) Lite in the [latest release](https://github.com/RaspAP/raspap-webgui/releases/tag/2.9.7).
